### DISCUSSION TITLE
Code Sever - Monitor productivity

### DISCUSSION THREAD

**Discussion by paulchill** [coder/code-server#5943] [Q&A]
Hey Guys,

I'm using code sever and love it! works great for our startup!

Looking to see if there is an easy to record developer usage activity.

eg. streams keystrokes or somethings simple shows when the dev last wrote code in vscode  etc

To know if a dev is actually writing code (regardless if quality or not), i'm well aware there are ways around this but seeing if there is any recommend simple solution, was thinking of streaming a log from the code-server or something to slack. 

Would love your feedback?

**Comment by lg-nineoneninethreenine** [comment]
Have you tried Wakatime? I think it could do the job for your requirement.

**Reply by jsjoeio** [reply]
Ha! I was about to recommend that as well. https://wakatime.com/vs-code

**Comment by paulchill** [comment]
is there any solution i can do without having to get any devs to install this plugin in code server them selves? that as an admin i can enforce?

**Reply by jsjoeio** [reply]
Not that I know of. cc @code-asher

**Reply by code-asher** [reply]
Hmm I think you would have to make your own modifications to code-server to track activity that closely.  The closest thing I can think of is to use telemetry which might give you a basic idea.
